src/adapter/src/catalog.rs

```diff
@@ -155,6 +156,7 @@ pub const LINKED_CLUSTER_REPLICA_NAME: &str = "linked";
 #[derive(Debug)]
 pub struct Catalog {
     state: CatalogState,
+    plans: CatalogPlans,
```

Comment:
It's unfortunate that this needs to live here because it has to be managed outside the transaction layer. I'm in favor of merging as is, and if someone from the adapter team wants to fix it (maybe figure out a way to integrate with parse_item) we can do the refactor.